### PR TITLE
Fixes some bugs

### PR DESCRIPTION
**Pull request by MarinaGryphon**
Fixes some bugs.

### CODE REVIEW

**Comment by MarinaGryphon** [comment]
!bugfix

**Comment by MarinaGryphon** [comment]
!review

**Comment by MarinaGryphon** [comment]
OK so for some reason if you join at roundstart as a sec officer Vaurca (only one I've tested) it dumps you into space on level 2. Whaaaat.

**Comment by MarinaGryphon** [comment]
nvm,,,

File: code/controllers/subsystems/air.dm [hunk 1]
```diff
@@ -423,15 +423,15 @@ Total Unsimulated Turfs: [world.maxx*world.maxy*world.maxz - simulated_turf_coun
 		return edge
 
 /datum/controller/subsystem/air/proc/has_same_air(turf/A, turf/B)
-	if(A.oxygen != B.oxygen) 
+	if(A?.oxygen != B?.oxygen)
```
Comment (skull132) [inline code/controllers/subsystems/air.dm]: Bad place for the `?.` operator. It's possible that _both_ are `null`, this results in everything passing and the proc STILL returning `1`. Simply add this as a front guard:
```dm
if (!A || !B)
    return 0
```

File: code/controllers/subsystems/job.dm [hunk 2]
```diff
@@ -409,9 +409,6 @@
 			var/obj/item/clothing/glasses/G = H.glasses
 			G.prescription = TRUE
 
-	if(H.species)
-		H.species.equip_later_gear(H)
```
Comment (skull132) [inline code/controllers/subsystems/job.dm]: Uuuh. Is this never being called now, or? (You should start describing your fixes in the PR, so I don't have to ask these questions.)

File: code/controllers/subsystems/ticker.dm [hunk 3]
```diff
@@ -613,20 +613,12 @@ var/datum/controller/subsystem/ticker/SSticker
 			minds += player.mind
 
 /datum/controller/subsystem/ticker/proc/equip_characters()
-	var/captainless = TRUE
```
Comment (skull132) [inline code/controllers/subsystems/ticker.dm]: This technically isn't a bugfix. 🤔

File: code/game/gamemodes/game_mode_latespawn.dm [hunk 4]
```diff
@@ -6,7 +6,7 @@
 ///process()
 ///Called by the gameticker
 /datum/game_mode/process()
-	if(round_autoantag && world.time >= next_spawn && !emergency_shuttle.departed)
+	if(round_autoantag && world.time >= next_spawn && emergency_shuttle.shuttle.moving_status == SHUTTLE_IDLE)
```
Comment (skull132) [inline code/game/gamemodes/game_mode_latespawn.dm]: This is not a fix at all. SHUTTLE_IDLE only stops the crew from being traitored while the shuttle is flying to the station, or back to CC. It isn't going to stop them from spawning as antag when:
* The first 6 minutes of the shuttle timer have elapsed.
* The shuttle is docked at the station.

File: code/controllers/subsystems/air.dm (hunk 1 above)
Comment (MarinaGryphon) [inline code/controllers/subsystems/air.dm]:
```suggestion
	if(A.oxygen != B.oxygen) 
```

File: code/controllers/subsystems/air.dm [hunk 5]
```diff
@@ -423,15 +423,15 @@ Total Unsimulated Turfs: [world.maxx*world.maxy*world.maxz - simulated_turf_coun
...
+	if(A?.nitrogen != B?.nitrogen) 
 		return 0
-	if(A.phoron != B.phoron) 
+	if(A?.phoron != B?.phoron) 
 		return 0
-	if(A.carbon_dioxide != B.carbon_dioxide)
+	if(A?.carbon_dioxide != B?.carbon_dioxide)
 		return 0
-	if(A.temperature != B.temperature)
+	if(A?.temperature != B?.temperature)
```
Comment (MarinaGryphon) [inline code/controllers/subsystems/air.dm]:
```suggestion
	if(A.temperature != B.temperature)
```

File: code/controllers/subsystems/air.dm [hunk 6]
```diff
@@ -423,15 +423,15 @@ Total Unsimulated Turfs: [world.maxx*world.maxy*world.maxz - simulated_turf_coun
...
+	if(A?.oxygen != B?.oxygen) 
 		return 0
-	if(A.nitrogen != B.nitrogen) 
+	if(A?.nitrogen != B?.nitrogen) 
 		return 0
-	if(A.phoron != B.phoron) 
+	if(A?.phoron != B?.phoron) 
 		return 0
-	if(A.carbon_dioxide != B.carbon_dioxide)
+	if(A?.carbon_dioxide != B?.carbon_dioxide)
```
Comment (MarinaGryphon) [inline code/controllers/subsystems/air.dm]:
```suggestion
	if(A.carbon_dioxide != B.carbon_dioxide)
```

File: code/controllers/subsystems/air.dm [hunk 7]
```diff
@@ -423,15 +423,15 @@ Total Unsimulated Turfs: [world.maxx*world.maxy*world.maxz - simulated_turf_coun
...
 
 /datum/controller/subsystem/air/proc/has_same_air(turf/A, turf/B)
-	if(A.oxygen != B.oxygen) 
+	if(A?.oxygen != B?.oxygen) 
 		return 0
-	if(A.nitrogen != B.nitrogen) 
+	if(A?.nitrogen != B?.nitrogen) 
 		return 0
-	if(A.phoron != B.phoron) 
+	if(A?.phoron != B?.phoron)
```
Comment (MarinaGryphon) [inline code/controllers/subsystems/air.dm]:
```suggestion
	if(A.phoron != B.phoron) 
```

File: code/controllers/subsystems/air.dm [hunk 8]
```diff
@@ -423,15 +423,15 @@ Total Unsimulated Turfs: [world.maxx*world.maxy*world.maxz - simulated_turf_coun
 		return edge
 
 /datum/controller/subsystem/air/proc/has_same_air(turf/A, turf/B)
-	if(A.oxygen != B.oxygen) 
+	if(A?.oxygen != B?.oxygen) 
 		return 0
-	if(A.nitrogen != B.nitrogen) 
+	if(A?.nitrogen != B?.nitrogen)
```
Comment (MarinaGryphon) [inline code/controllers/subsystems/air.dm]:
```suggestion
	if(A.nitrogen != B.nitrogen) 
```

File: code/controllers/subsystems/job.dm (hunk 2 above)
Comment (MarinaGryphon) [reply code/controllers/subsystems/job.dm]: It's called in a different proc. This meant it was called twice, iirc.

File: code/controllers/subsystems/ticker.dm (hunk 3 above)
Comment (MarinaGryphon) [reply code/controllers/subsystems/ticker.dm]: it's annoying though, and served no functionality; one could argue that it is fixing something that bugs me, therefore it is a bug :^)

File: code/game/gamemodes/game_mode_latespawn.dm (hunk 4 above)
Comment (MarinaGryphon) [reply code/game/gamemodes/game_mode_latespawn.dm]: AAAAAAAAAAAAAUGH okay. How would one fix that.

File: code/controllers/subsystems/job.dm (hunk 2 above)
Comment (MarinaGryphon) [reply code/controllers/subsystems/job.dm]: Line 550 in this file after the changes.

**Comment by Kaedwuff** [comment]
Why has this been sitting in awaiting merge since october 26?

If there are still changes required shouldn't it be in that tag state?

**Comment by NonQueueingMatt** [comment]
Closing, stale PR.

**Pull request closed without merging**